### DISCUSSION TITLE
How to implements 'this' keyword in with langium ?

### DISCUSSION THREAD

**Discussion by hyphappy011** [eclipse-langium/langium#1325] [Q&A] [answered]
I copy the[ langium-lox](https://github.com/TypeFox/langium-lox) example, and write some code with a simple class in the playground, but it can not deal the "this" keyword. 
How can i fix it?

```ts
class testClass {
  name: string

  testIt():void {
     this.name = "123";
  }
}
```

![image](https://github.com/eclipse-langium/langium/assets/24590408/d31b46d0-e31a-40a3-89a7-c675c098b01c)

**Comment by msujew** [comment, marked as answer]
Hey @hyphappy011,

to support the `this` keyword correctly, you would need to implement a specialized scope provider. You can find a general guide on it [here](https://langium.org/guides/scoping/class-member/). Unfortunately, the Langium playground doesn't offer to define a new scope provider, and the default scoping doesn't cover this kind of advanced use case.
